components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/APIConsumerImpl.java

```diff
@@ -3299,7 +3300,50 @@ public void deleteOAuthApplication(String consumerKey) throws APIManagementExcep
         }
     }
 
-	public JSONObject resumeWorkflow(Object[] args) {
+    @Override
+    public Application[] getApplicationsByOwner(String userId) throws APIManagementException {
+        return apiMgtDAO.getApplicationsByOwner(userId);
+    }
+
+    @Override
+    public boolean updateApplicationOwner(String userId, Application application) throws APIManagementException {
+
+        boolean isAppUpdated = false;
+
+        try {
+            String oldUserRoles[];
+            RealmService realmService = ServiceReferenceHolder.getInstance().getRealmService();
+            int tenantId = ServiceReferenceHolder.getInstance().getRealmService().getTenantManager()
+                    .getTenantId(MultitenantUtils.getTenantDomain(username));
+            UserStoreManager userStoreManager = realmService.getTenantUserRealm(tenantId).getUserStoreManager();
+            String oldUserName = application.getSubscriber().getName();
+            oldUserRoles = userStoreManager.getRoleListOfUser(MultitenantUtils.getTenantAwareUsername(oldUserName));
+
+            for (String role : oldUserRoles) {
+                if (role.contains(application.getName())) {
+                    userStoreManager.updateRoleListOfUser(userId, null, new String[] {role});
+                }
+            }
+            isAppUpdated = true;
+
+        } catch (org.wso2.carbon.user.api.UserStoreException e) {
+            // Log and return since we do not want to update the APIM datbase if we cannot assign the roles to the
+            // new user
+            isAppUpdated = false;
+            log.error("Error when getting the tenant's UserStoreManager or when getting roles of user ", e);
```

Comment:
without using a flag to indicate whether the operation was successful/not, we can handle the exception from the parent level.

Reply:
Fixed it